xds/src/main/java/io/grpc/xds/internal/sds/SdsProtocolNegotiators.java

```diff
@@ -294,11 +293,29 @@ public void handlerAdded(ChannelHandlerContext ctx) throws Exception {
           });
       checkNotNull(grpcHandler, "grpcHandler");
       this.grpcHandler = grpcHandler;
-      this.downstreamTlsContext = downstreamTlsContext;
+      this.downstreamTlsContextFromBuilder = downstreamTlsContext;
+      this.xdsClientWrapperForServerSds = xdsClientWrapperForServerSds;
+    }
+
+    private static boolean isTlsContextEmpty(DownstreamTlsContext downstreamTlsContext) {
+      return downstreamTlsContext == null || !downstreamTlsContext.hasCommonTlsContext();
     }
 
     @Override
     protected void handlerAdded0(final ChannelHandlerContext ctx) {
```

Comment:
we would love to make ProtocolNegotiationHandler as small as possible. can you make a new handler that decides which negotiator to use (plaintext/sds)?

Reply:
Sounds like a good idea. How about if I create a TODO task for a separate PR? So this PR doesn't have too many changes.

Reply:
this should be relatively simple change (okay maybe some boiler plates, but this way will be helpful when you need to debug), i don't see why we want to delay.

Reply:
Yes, it is actually not that difficult. And with your other suggestion about handling exception(s) in the caller for `xdsClientWrapperForServerSds`, this change goes well with that. So I'll do it in this one